admin_tests/regisration_schemas/test_views.py

```diff
@@ -0,0 +1,198 @@
+        plain_view = views.RegistrationSchemaCreateView()
+        view = setup_view(plain_view, req)
+        return view
+
+    def test_registration_schema_create(self, view, csv_file, req):
+        # django.contrib.messages has a bug which effects unittests
+        # more info here -> https://code.djangoproject.com/ticket/17971
+        setattr(req, 'session', 'session')
+        messages = FallbackStorage(req)
+        setattr(req, '_messages', messages)
+
+        form = view.get_form()
+
+        form.data['name'] = 'Trust the Process'
+        form.files['schema'] = csv_file
+
+        view.form_valid(form)
+
+        registration_schema = RegistrationSchema.objects.get(name='Trust the Process')
+        assert registration_schema.schema_blocks.count() == 1
+        block = registration_schema.schema_blocks.first()
+        assert block.block_type == 'page-heading'
+        assert block.display_text == 'This is the page heading'
+        assert registration_schema.schema_version == 1
+
+        view.form_valid(form)
+        registration_schema = RegistrationSchema.objects.get_latest_version(name='Trust the Process')
+        assert registration_schema.schema_version == 2
+
+    def test_registration_schema_csv_to_blocks(self, view, csv_file):
+        blocks = view.csv_to_blocks(csv_file)
+        assert len(blocks) == 1
+        assert blocks[0]['block_type'] == 'page-heading'
+        assert blocks[0]['display_text'] == 'This is the page heading'
+
+
+@pytest.mark.django_db
+@pytest.mark.urls('admin.base.urls')
+class TestDeleteRegistrationSchema:
+
+    @pytest.fixture()
+    def req(self):
+        return RequestFactory().get('/fake_path')
+
+    @pytest.fixture()
+    def registration_schema(self):
+        return RegistrationSchema.objects.create(
+            name='foo',
+            schema={'foo': 42, 'atomicSchema': True},
+            schema_version=1,
+            active=False,
+            visible=False
+        )
+
+    @pytest.fixture()
+    def provider(self, registration_schema):
+        provider = RegistrationProviderFactory()
+        registration_schema.providers.add(provider)
+        return provider
+
+    @pytest.fixture()
+    def view(self, req, registration_schema):
+        view = views.RegistrationSchemaDeleteView()
+        view = setup_view(view, req)
+        view.kwargs = {'registration_schema_id': registration_schema.id}
+        return view
+
+    def test_registration_schema_delete(self, req, view, registration_schema):
+        # django.contrib.messages has a bug which effects unittests
+        # more info here -> https://code.djangoproject.com/ticket/17971
+        setattr(req, 'session', 'session')
+        messages = FallbackStorage(req)
+        setattr(req, '_messages', messages)
+
+        view.delete(req)
+        assert not RegistrationSchema.objects.filter(id=registration_schema.id)
+
+    def test_registration_schema_delete_validate(self, req, view, registration_schema, provider):
```

Comment:
nit: Name should be clearer about the fact that we expect this to fail (and why)